libs/nio/src/main/java/org/elasticsearch/nio/NioSelector.java

```diff
@@ -452,28 +477,9 @@ private void handleQueuedWrites() {
         }
     }
 
-    /**
-     * This is a convenience method to be called after some object (normally channels) are enqueued with this
-     * selector. This method will check if the selector is still open. If it is open, normal operation can
-     * proceed.
-     *
-     * If the selector is closed, then we attempt to remove the object from the queue. If the removal
-     * succeeds then we throw an {@link IllegalStateException} indicating that normal operation failed. If
-     * the object cannot be removed from the queue, then the object has already been handled by the selector
-     * and operation can proceed normally.
-     *
-     * If this method is called from the selector thread, we will not allow the queuing to occur as the
-     * selector thread can manipulate its queues internally even if it is no longer open.
-     *
-     * @param queue the queue to which the object was added
-     * @param objectAdded the objected added
-     * @param <O> the object type
-     */
-    private <O> void ensureSelectorOpenForEnqueuing(ConcurrentLinkedQueue<O> queue, O objectAdded) {
-        if (isOpen() == false && isOnCurrentThread() == false) {
-            if (queue.remove(objectAdded)) {
+    private void ensureSelectorOpenForEnqueuing() {
```

Comment:
This work needs to come back. It is still relevant. If the enqueued operation has been removed then we should not throw the exception. It is fine to remove the isOnCurrentThread check.

Reply:
Right ... sorry about that, reverting.